test/sql/multi_transaction_index.sql

```diff
@@ -0,0 +1,193 @@
+
+-- Explicitly defining an index tablespace should work and propagate
+-- to chunks
+CREATE INDEX ON index_test (time, device) WITH (hypertable.multi_transaction) TABLESPACE tablespace2;
+
+-- New indexes without explicit tablespaces should use the default
+-- tablespace
+CREATE INDEX ON index_test (device) WITH (hypertable.multi_transaction);
+
+-- Create chunk
+INSERT INTO index_test VALUES ('2017-01-20T09:00:01', 17.5);
+
+-- Check that the tablespaces of chunk indexes match the tablespace of
+-- the main index
+SELECT * FROM test.show_indexes('index_test');
+SELECT * FROM test.show_indexesp('_timescaledb_internal._hyper%_chunk');
+
+-- Creating a new index should propagate to existing chunks, including
+-- the given tablespace
+CREATE INDEX ON index_test (time, temp) WITH (hypertable.multi_transaction) TABLESPACE tablespace2;
+
+SELECT * FROM test.show_indexes('index_test');
+SELECT * FROM test.show_indexesp('_timescaledb_internal._hyper%_chunk');
+
+-- Cleanup
+DROP TABLE index_test CASCADE;
+\c :TEST_DBNAME :ROLE_SUPERUSER
+DROP TABLESPACE tablespace1;
+DROP TABLESPACE tablespace2;
+
+-- Test that indexes are planned correctly
+CREATE TABLE index_expr_test(id serial, time timestamptz, temp float, meta int);
+select create_hypertable('index_expr_test', 'time');
+
+-- Screw up the attribute numbers
+ALTER TABLE index_expr_test DROP COLUMN id;
+
+CREATE INDEX ON index_expr_test (meta) WITH (hypertable.multi_transaction);
+INSERT INTO index_expr_test VALUES ('2017-01-20T09:00:01', 17.5, 1);
+INSERT INTO index_expr_test VALUES ('2017-01-20T09:00:01', 17.5, 2);
+
+
+SET enable_seqscan TO false;
+SET enable_bitmapscan TO false;
+EXPLAIN (verbose, costs off)
+SELECT * FROM index_expr_test WHERE meta = 1;
+SELECT * FROM index_expr_test WHERE meta = 1;
+SET enable_seqscan TO default;
+SET enable_bitmapscan TO default;
+
+\set ON_ERROR_STOP 0
+-- cannot create a multi_transaction index within a transaction block
+BEGIN;
+CREATE INDEX ON index_expr_test (temp) WITH (hypertable.multi_transaction);
+ROLLBACK;
+\set ON_ERROR_STOP 1
+
+DROP TABLE index_expr_test CASCADE;
+
+CREATE TABLE partial_index_test(time INTEGER);
+SELECT create_hypertable('partial_index_test', 'time', chunk_time_interval => 1, create_default_indexes => false);
+
+-- create 3 chunks
+INSERT INTO partial_index_test(time) SELECT generate_series(0, 2);
+
+select * from partial_index_test order by 1;
+
+-- create indexes on only 1 of the chunks
+CREATE INDEX ON partial_index_test (time) WITH (hypertable.multi_transaction, hypertable.max_chunks='1');
+
+SELECT * FROM test.show_indexes('partial_index_test');
+SELECT * FROM test.show_indexesp('_timescaledb_internal._hyper%_chunk');
+
+-- with broken gapfill, this caused an assertion failure
+SET enable_seqscan TO false;
+SET enable_bitmapscan TO false;
+EXPLAIN (verbose, costs off) SELECT * FROM partial_index_test WHERE time < 2 ORDER BY time LIMIT 2;
+SELECT * FROM partial_index_test WHERE time < 2 ORDER BY time LIMIT 2;
```

Comment:
This is the test case that failed before PR #1008

Reply:
maybe add comment that this will result in MergeAppendPath with children with non-matching pathkeys

Reply:
this hit an assertion in ordered append not gapfill